### DISCUSSION TITLE
Clean install on macOS - No music

### DISCUSSION THREAD

**Discussion by cbryson84** [CorsixTH/CorsixTH#2570] [General Diagnostics] [answered]
Hello,

I've installed CorsixTH and have no music. When I launch from the terminal I'm seeing the errors shown below. I purchased on GOG and moved the "game" folder into my Documents. Do I need to convert the files or something?

Thanks,
Chris.

macOS 14.4.1 on a 14" M1 Pro

```---------------------------------------------------------------

Welcome to CorsixTH v0.67!

---------------------------------------------------------------

This window will display useful information if an error occurs.

---------------------------------------------------------------

An error has occurred!
Almost anything can be the cause, but the detailed information below can help the developers find the source of the error.
Running: The callback handler.
A stack trace is included below, and the handler has been disconnected.
...1/CorsixTH.app/Contents/MacOS/../Resources/Lua/audio.lua:583: Could not load music file '/Users/chris/Documents/Theme Hospital Game Files/game/SOUND/MIDI/NIGHTSH.XMI' (Unrecognized file type (not AIFF or AIFC))
stack traceback:
	[C]: in function 'coroutine.yield'
	...7_1/CorsixTH.app/Contents/MacOS/../Resources/Lua/app.lua:1102: in function <...7_1/CorsixTH.app/Contents/MacOS/../Resources/Lua/app.lua:1097>

Cannot check for updates since LuaSocket and/or LuaSec are not available.
An error has occurred!
Almost anything can be the cause, but the detailed information below can help the developers find the source of the error.
Running: The callback handler.
A stack trace is included below, and the handler has been disconnected.
...1/CorsixTH.app/Contents/MacOS/../Resources/Lua/audio.lua:583: Could not load music file '/Users/chris/Documents/Theme Hospital Game Files/game/SOUND/MIDI/STEADY.XMI' (Unrecognized file type (not AIFF or AIFC))
stack traceback:
	[C]: in function 'coroutine.yield'
	...7_1/CorsixTH.app/Contents/MacOS/../Resources/Lua/app.lua:1102: in function <...7_1/CorsixTH.app/Contents/MacOS/../Resources/Lua/app.lua:1097>```

**Comment by tobylane** [comment]
This is a combination of a few issues we've recently fixed. Can you try a recent build, either your own or the artifact [here.](https://github.com/tobylane/CorsixTH/actions/runs/9035743503#artifacts)

**Comment by cbryson84** [comment]
Thank you for the prompt response. I'm running v0.67 installed via Homebrew - is this not the latest build?

Pardon my ignorance, homebrew was fairly new to me.

**Reply by lewri** [reply]
Could their Homebrew-core formula need to be adjusted maybe?
https://github.com/Homebrew/homebrew-core/blob/b55d515d2f41107490acd0bff314cd42adaacd11/Formula/c/corsixth.rb

Failing that, we now sign our MacOS builds if you prefer that.

**Reply by tobylane** [reply]
You are on the latest release, you can `brew install --head corsixth` to test the development branch. One fix we have made is about xmi files, which is probably the solution. 

Homebrew puts everything in sdl2 mixer, more than vcpkg. In theory they could add a few more file types to the music extensions set, but they won't.

**Comment by cbryson84** [comment]
I have completely removed the original CorsixTH, installed XCode and ran the `brew install --head corsixth` but unfortunately I'm still getting the same result. Could it be another issue? Is there a workaround I can attempt?

```

---------------------------------------------------------------

Welcome to CorsixTH Trunk!

---------------------------------------------------------------

This window will display useful information if an error occurs.

---------------------------------------------------------------

Configured unicode font not found, using /Library/Fonts/Arial Unicode.ttf instead.
This will be written to the config file.
An error has occurred!
Almost anything can be the cause, but the detailed information below can help the developers find the source of the error.
Running: The callback handler.
A stack trace is included below, and the handler has been disconnected.
...1/CorsixTH.app/Contents/MacOS/../Resources/Lua/audio.lua:595: Could not load music file 'Sound/Midi/FORTUNE.XMI' (No SoundFonts have been requested)
stack traceback:
	[C]: in function 'coroutine.yield'
	...c_1/CorsixTH.app/Contents/MacOS/../Resources/Lua/app.lua:1139: in function <...c_1/CorsixTH.app/Contents/MacOS/../Resources/Lua/app.lua:1134>

Will not check for updates since this is the Trunk version.
An error has occurred!
Almost anything can be the cause, but the detailed information below can help the developers find the source of the error.
Running: The callback handler.
A stack trace is included below, and the handler has been disconnected.
...1/CorsixTH.app/Contents/MacOS/../Resources/Lua/audio.lua:595: Could not load music file 'Sound/Midi/STEADY.XMI' (No SoundFonts have been requested)
stack traceback:
	[C]: in function 'coroutine.yield'
	...c_1/CorsixTH.app/Contents/MacOS/../Resources/Lua/app.lua:1139: in function <...c_1/CorsixTH.app/Contents/MacOS/../Resources/Lua/app.lua:1134>```

**Reply by lewri** [reply]
It's a different error now, it's missing the soundfont FluidR3. I'm not sure where it goes though... @tobylane ?

Edit: actually doesn't macos try to use its built-in?

**Reply by tobylane** [reply]
This is using fluid synth for midi, rather than native. You can download this file to, I think, the Resources folder. https://raw.githubusercontent.com/Jacalz/fluid-soundfont/master/SF3/FluidR3.sf3

**Reply by cbryson84** [reply]
That done it! Thank you!

**Reply by lewri** [reply]
Oh, has this been run without this command (in our how to compile)?
`brew install --with-mpg123 --with-flac --with-libmikmod --with-fluid-synth https://raw.githubusercontent.com/Homebrew/homebrew-core/a9114e/Formula/sdl2_mixer.rb`
^ Edit: above looks outdated to how other brews look

**Reply by cbryson84** [reply, marked as answer]
I used the command `brew install --head corsixth` maybe that was the issue? All seems to be good now though.
